Us4 details page

gwittm opened the pull request.

NadjaKanunnikow commented on components/StyledDetailsPage.js:
```diff
@@ -0,0 +1,28 @@
+import styled from "styled-components";
+
+export const HeaderDiv = styled.div`
```
For us it would be more semantic to style a `<h2>` instead of a `<div>`element. 
Similarly, we would, for semantic reasons, suggest to use "Heading" instead of "Header".

NadjaKanunnikow commented on pages/recipes/[id]/index.jsx:
```diff
@@ -0,0 +1,54 @@
+            <div key={index}>
+              {ingredient.name}
+              {":"} {ingredient.amount} {ingredient.unit}
+            </div>
```
```suggestion
            </li>
```

NadjaKanunnikow commented on pages/recipes/[id]/index.jsx:
```diff
@@ -0,0 +1,54 @@
+      <StyledDetailsItem>
+        <h4>Ingredients:</h4>
+        <ul>
+          {recipe.ingredients.map((ingredient, index) => (
+            <div key={index}>
```
As we recently learned from @doemser using the index as a key might lead to problems with react. 
Compare: https://github.com/orgs/neuefische/discussions/775

Creating and using an unique id would be a better option.

NadjaKanunnikow reviewed (changes requested):
We noticed that you use both pixel and rem as a unit. To make your code more consistent and responsive use as much rem as possible.

gwittm commented:
Thank you for your comments!
We changed the divs into li-items. 
Pixel and rems will be changed as soon as we create our final styling including fixed paddings, margins... instead of touching everything several times ;-)
Regarding the index: in our case we think the index is correct for this userstory, as we just list all added ingredients in the same order we added them before...

NikRoe commented on components/Card.js:
```diff
@@ -0,0 +1,10 @@
+import Link from "next/link.js";
+import styled from "styled-components";
```
you don't need this import

NikRoe commented on components/Card.js:
```diff
@@ -0,0 +1,10 @@
+import Link from "next/link.js";
+import styled from "styled-components";
+
+export default function Card({ title, id }) {
```
this isn't really a card, but a link 🤔

NikRoe commented on components/Card.js:
```diff
@@ -0,0 +1,10 @@
+
+export default function Card({ title, id }) {
+  return (
+    <Link href={`recipes/${id}`} passHref legacyBehavior>
+      <p>{title}</p>
+    </Link>
```
this is even better and if you want the Link the be a block element, you can style it that way

```suggestion
    <Link href={`recipes/${id}`}>
     {title}
    </Link>
```

gwittm commented on pages/recipes/[id]/index.jsx:
```diff
@@ -0,0 +1,54 @@
+      <br />
+      <Link href={"/"} passHref legacyBehavior>
+        <StyledLink $justifySelf="start">back</StyledLink>
+      </Link>
```
hope we got that point correctly ;-)

NikRoe commented on pages/recipes/[id]/index.js:
```diff
@@ -0,0 +1,55 @@
+      <StyledDetailsItem>
+        <h4>Ingredients:</h4>
+        <ul>
+          {recipe.ingredients.map((ingredient, ingredientID) => (
+            <li key={ingredientID}>
```
the second parameter of .map() is always the index of the current element being processed in the array. Regardless of how you name it.

But your ingredient object has a key id and a key _id that are both unique and that you can and should use here instead:

```suggestion
          {recipe.ingredients.map((ingredient) => (
            <li key={ingredient.id}>
```

NikRoe commented on pages/index.js:
```diff
@@ -1,28 +1,44 @@
 import Link from "next/link.js";
 import styled from "styled-components";
 import useSWR from "swr";
+/* import { StyledLink } from "@/components/StyledLink"; */
```
```suggestion
```

NikRoe reviewed (changes requested):
Hi there,

you are almost there.
The change of your key did not achieve what you intended. I left a suggestion how to implement it instead.

Best regards 👋

gwittm commented:
thanks for your comments. We changed it :)

gregorsart commented on pages/recipes/[id]/index.js:
```diff
@@ -0,0 +1,54 @@
+
+  return (
+    <StyledDetailsPageContainer>
+      <HeaderDiv>
```
If this is your Header of the Website, you should give it the `header` semantic name instead of `div`.

gwittm replied:
just changed, thank you :)
